core/src/test/java/fr/sncf/osrd/envelope_physics/TrainPhysics.java

```diff
@@ -0,0 +1,123 @@
+package fr.sncf.osrd.envelope_physics;
+
+import static fr.sncf.osrd.envelope_sim.TrainPhysicsIntegrator.getWeightForce;
+import static fr.sncf.osrd.envelope_sim.TrainPhysicsIntegrator.newtonStep;
+import static org.junit.jupiter.api.Assertions.*;
+
+import fr.sncf.osrd.envelope_sim.Action;
+import fr.sncf.osrd.envelope_sim.EnvelopeSimContext;
+import fr.sncf.osrd.envelope_sim.FlatPath;
+import fr.sncf.osrd.envelope_sim.TrainPhysicsIntegrator;
+import fr.sncf.osrd.train.RollingStock;
+import fr.sncf.osrd.train.TestTrains;
+import org.junit.jupiter.api.Test;
+
+
+public class TrainPhysics {
+
+    RollingStock testRollingStock = TestTrains.REALISTIC_FAST_TRAIN;
+    double timeStep = 1.0;
+
+    @Test
+    public void testSlopeNoTraction() {
+        var testPath = new FlatPath(100000, -40);
+        var context = new EnvelopeSimContext(testRollingStock, testPath, timeStep);
+        double position = 0.0;
+        double speed = 0.0;
+
+        // how fast would a train go after 10 steps of 1 sec, coasting on a
+        // -40m / km slope?
+        for (int i = 0; i < 10; i++) {
+            var step = TrainPhysicsIntegrator.step(context, position, speed, Action.COAST, +1);
+            position += step.positionDelta;
+            speed = step.endSpeed;
+        }
+
+        // we expect about +4m/s (the train goes forward)
+        assertTrue(speed < +5 && speed > +1, String.valueOf(speed));
+    }
+
+    @Test
+    public void testSteepSlopeTraction() {
+        var testPath = new FlatPath(100000, -45);
+        var context = new EnvelopeSimContext(testRollingStock, testPath, timeStep);
+        double position = 0.0;
+        double speed = 0.0;
+
+        // how fast would a train go after 10 steps of 1 sec, full throttle on a 45deg slope?
+        for (int i = 0; i < 10; i++) {
+            var step = TrainPhysicsIntegrator.step(context, position, speed, Action.ACCELERATE, +1);
+            position += step.positionDelta;
+            speed = step.endSpeed;
+        }
+        // we expect the train to go pretty fast
+        assertTrue(speed > 8 && speed < 100, String.valueOf(speed));
```

Comment:
same here, knowing the train characteritics, we can assert that the final speed is in a shorter range than [8, 100].
again i know this is how the test was built originally

Reply:
solved by https://github.com/DGEXSolutions/osrd/pull/636/commits/5b9573d3ce5ed5d2acf1fa96f4b8d4d5d19e01d8